packages/terra-functional-testing/src/services/wdio-terra-service/WDIOTerraService.js

```diff
@@ -0,0 +1,280 @@
+const expect = require('expect');
+const fs = require('fs-extra');
+const path = require('path');
+const { SevereServiceError } = require('webdriverio');
+const { accessibility, element, screenshot } = require('../../commands/validates');
+const { toBeAccessible, toMatchReference } = require('../../commands/expect');
+const {
+  describeTests,
+  describeViewports,
+  getViewports,
+  hideInputCaret,
+  ScreenshotRequestor,
+  setApplicationLocale,
+  setViewport,
+} = require('../../commands/utils');
+const { BUILD_BRANCH, BUILD_TYPE } = require('../../constants');
+const GithubIssue = require('./GithubIssue');
+const GithubPr = require('./GithubPr');
+
+class WDIOTerraService {
+  /**
+   * Service constructor.
+   * @param {Object} _options - The options specific to this service.
+   * @param {Object} _capabilities - The list of capabilities details.
+   * @param {Object} config - The object containing the wdio configuration and options defined in the terra-cli test runner.
+   */
+  constructor(_options, _capabilities, config = {}) {
+    const { serviceOptions, launcherOptions } = config;
+
+    this.serviceOptions = {
+      selector: '[data-terra-test-content] *:first-child',
+      ...launcherOptions,
+      ...serviceOptions,
+    };
+    this.getRemoteScreenshotConfiguration = config.getRemoteScreenshotConfiguration;
+    this.screenshotsSites = config.screenshotsSites;
+  }
+
+  /**
+   * Parses Package.json for metadata.
+   * @returns An object of metadata.
+   */
+  static async getRepoMetadata() {
+    const metadata = await fs.readJson(path.join(process.cwd(), 'package.json'));
+
+    // Do not try to use the URL constructor here. It will fail on urls with
+    // custom protocols like 'git+https://'.
+    const [, rOwner, rName] = metadata.repository.url.split('://')[1].split('/');
+
+    return {
+      owner: rOwner,
+      name: rName,
+    };
+  }
+
+  /**
+   * Gets executed once before all workers get launched.
+   * Downloads the reference screenshots from the remote repository if useRemoteReferenceScreenshots is true.
+   */
+  async onPrepare() {
+    const {
+      buildBranch,
+      gitApiUrl,
+      gitToken,
+      issueNumber,
+      useRemoteReferenceScreenshots,
+    } = this.serviceOptions;
+
+    if (!useRemoteReferenceScreenshots) {
+      return;
+    }
+
+    let screenshotConfig;
+    try {
+      if (buildBranch.match(BUILD_BRANCH.pullRequest)) {
+        // When building PR branches the reference screenshots are downloaded from the PR's base branch's remote repo.
+        const metadata = await WDIOTerraService.getRepoMetadata();
+        const gp = new GithubPr(
+          gitApiUrl,
+          gitToken,
+          metadata.owner,
+          metadata.repo,
+          issueNumber,
+        );
+
+        const baseBranch = await gp.getBaseBranchRef();
+        screenshotConfig = this.getRemoteScreenshotConfiguration(this.screenshotsSites, baseBranch);
```

Comment:
Should we add back this [check for getRemoteScreenshotConfiguration](https://github.com/cerner/terra-toolkit/pull/764/commits/82348fcecf4085e1a672cd79259e01464d17930d#diff-a6ba3c1d16c08e58e7ee3994f3ab541b63b2a3a8dafb0de9daed2c8090c948d9R66-R68) in case it is not configured correctly in the consumer's wdio.config.js?

Reply:
Nevermind. This is all in a try/catch block so it would catch if there is an error.